posthog/temporal/data_modeling/ducklake_copy_workflow.py

```diff
@@ -163,123 +173,131 @@ def verify_ducklake_copy_activity(inputs: DuckLakeCopyActivityInputs) -> list[Du
         logger.info("No DuckLake verification queries configured - skipping")
         return []
 
-    config = get_config()
-    conn = duckdb.connect()
-    alias = "ducklake_dev"
-    results: list[DuckLakeCopyVerificationResult] = []
+    heartbeater = HeartbeaterSync(details=("ducklake_verify", inputs.model.model_label), logger=logger)
+    with heartbeater:
+        config = get_config()
+        conn = duckdb.connect()
+        alias = "ducklake_dev"
+        results: list[DuckLakeCopyVerificationResult] = []
 
-    try:
-        _configure_source_storage(conn, logger)
-        configure_connection(conn, config, install_extension=True)
-        _attach_ducklake_catalog(conn, config, alias=alias)
-
-        ducklake_table = f"{alias}.{inputs.model.schema_name}.{inputs.model.table_name}"
-        format_values = {
-            "ducklake_table": ducklake_table,
-            "ducklake_schema": f"{alias}.{inputs.model.schema_name}",
-            "ducklake_alias": alias,
-            "schema_name": inputs.model.schema_name,
-            "table_name": inputs.model.table_name,
-        }
-
-        for query in inputs.model.verification_queries:
-            rendered_sql = query.sql.format(**format_values)
-            params = [_resolve_verification_parameter(param, inputs) for param in query.parameters]
-
-            try:
-                row = conn.execute(rendered_sql, params).fetchone()
-            except Exception as exc:
-                logger.warning(
-                    "DuckLake verification query failed",
-                    check=query.name,
-                    error=str(exc),
-                )
-                results.append(
-                    DuckLakeCopyVerificationResult(
-                        name=query.name,
-                        passed=False,
-                        expected_value=query.expected_value,
-                        tolerance=query.tolerance,
-                        description=query.description,
-                        sql=rendered_sql,
+        try:
+            heartbeater.details = ("verify_configure_source", inputs.model.model_label)
```

Comment:
You don't need to pass any details into the heartbeat unless if you're reading them later on - this isn't stored anywhere nor is it accessible via temporal UI I believe. It just allows you to pass vars between activity (re)attempts